thorlcr/master/thmastermain.cpp

```diff
@@ -314,74 +326,73 @@ class CRegistryServer : public CSimpleInterface
                 }
-                StringBuffer epStr;
-                PROGLOG("Slave %u connected from %s", slaveNum, sender->endpoint().getEndpointHostText(epStr).str());
                 --remaining;
             }
         }
-        assertex(slaves == connectedSlaves.ordinality());
-
-        if (isContainerized())
-        {
-            unsigned wfid = globals->getPropInt("@wfid");
-            const char *wuid = globals->queryProp("@workunit");
-            const char *graphName = globals->queryProp("@graphName");
-            Owned<IWorkUnitFactory> factory = getWorkUnitFactory();
-            Owned<IWorkUnit> workunit = factory->updateWorkUnit(wuid);
-            addTimeStamp(workunit, wfid, graphName, StWhenK8sReady);
-            publishPodNames(workunit, graphName);
-        }
+        assertex(slaves == connectedWorkers.size());
 
         unsigned localThorPortInc = globals->getPropInt("@localThorPortInc", DEFAULT_SLAVEPORTINC);
         unsigned slaveBasePort = globals->getPropInt("@slaveport", DEFAULT_THORSLAVEPORT);
         unsigned channelsPerWorker = globals->getPropInt("@channelsPerWorker", 1);
 
         Owned<IGroup> processGroup;
 
-        // NB: in bare metal Thor is bound to a group and cluster/communicator have alreday been setup (see earlier setClusterGroup call)
+        // NB: in bare metal Thor is bound to a group and cluster/communicator have already been setup (see earlier setClusterGroup call)
         if (clusterInitialized())
             processGroup.set(&queryProcessGroup());
         else
         {
-            /* sort by {port, ip}
-             * So that workers are not bunched on same node, but striped across the pod ips
-             */
-            auto compareINodeOrder = [](IInterface * const *ll, IInterface * const *rr)
+            if (isContainerized())
             {
-                INode *l = (INode *) *ll;
-                INode *r = (INode *) *rr;
-                const SocketEndpoint &lep = l->endpoint();
-                const SocketEndpoint &rep = r->endpoint();
-                if (lep.port < rep.port)
-                    return -1;
-                else if (lep.port > rep.port)
-                    return 1;
-                return lep.ipcompare(rep);
-            };
-            connectedSlaves.sort(compareINodeOrder);
-            processGroup.setown(createIGroup(connectedSlaves.ordinality(), connectedSlaves.getArray()));
+                // sort by pod+container name so that storage striping doesn't clump too much within single pods
+                auto sortFunc = [](const ConnectedWorkerDetail& a, const ConnectedWorkerDetail& b)
+                {
+                    if (std::get<1>(a) != std::get<1>(b))
+                        return std::get<1>(a) < std::get<1>(b);
+                    return std::get<2>(a) < std::get<2>(b);
+                };
+                std::sort(connectedWorkers.begin(), connectedWorkers.end(), sortFunc);
+            }
```

Comment:
else?  Should there be similar code, or is the old code redundant?  Is the isContainerized() test redundant - must it always be true?